CMake/Runtime/DownloadCacher.cmake

```diff
@@ -0,0 +1,113 @@
+#
+#  DownloadCacher.cmake
+#  Copyright 2021 ItJustWorksTM
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+
+# function for cleaning the cache
+function (clear_download_cache)
+  # delete all contents in the cached folder
+  file (REMOVE_RECURSE "${SMCE_DIR}/cached_downloads/")
+  message ("Cache has been cleared!")
+endfunction (clear_download_cache)
+
+#[================================================================================================================[.rst:
+cached_download
+--------------------
+
+Function to allow for caching of various downloads.
+
+Usage:
+.. code-block:: cmake
+
+  cached_download(<URL> <DEST> <RESULT_VARIABLE> <FORCE_UPDATE> )
+
+Where ``<URL>`` is the URL to the file to be downloaded, ``<DEST>`` is the name of the variable to store the absolute
+real path to the download location passed to the parent scope by the function, ``<RESULT_VARIABLE>``
+is the name of the variable to store the result of all processes passed to the parent scope by
+the function, ``<FORCE_UPDATE>`` will define whether an already cached download should be re-downloaded and cached.
+
+
+Note:
+No additional arguments except for the ones defined are allowed in the function call.
+Uses SHA256 to create a uniquely hashed download location for each download.
+File is locked until each download process is completed.
```

Comment:
This is ambiguous

Reply:
ambiguous how? if it's vague I could just remove it

Reply:
Doesn't say which file is locked nor what kind of "process" you're talking about

Reply:
I revised it, is gut?

Reply:
Still gotta update the description to reflect the placeholders in the call syntax